File: extract_DE.m
%% 

clear,clc

% 读取数据就是SEED数据集下的ExtractFeatures文件夹，但是需要把文件分出session，分别放在session1 2 3下面
load_path = './SEED/ExtractedFeatures1/session3/';
save_path = './SEED/SEED_code/DE/session3/';
label_path = './SEED/ExtractedFeatures1/label';
load(label_path)
file_list = dir(load_path);
file_list(1:2)=[];

for sub_i = 1:length(file_list)
    disp(['subject>',file_list(sub_i).name])
    S = load([load_path,file_list(sub_i).name]);
    DE = [];
    labelAll = [];
    for ii = 1:15
        eval(['data=','S.de_LDS',num2str(ii),';']);
        DE = cat(2,DE,data);
        labelAll = [labelAll;zeros(size(data,2),1)+label(ii)];
    end
    
    save([save_path,file_list(sub_i).name],'DE','labelAll');
    
end
    

File: extract_DE.py
import scipy.io
import os
import numpy as np

extrac_path = './SEED/SEED_code/ExtractedFeatures/session1/'
label_path = './SEED/SEED_code/ExtractedFeatures/session1/label.mat'
save_path = './SEED/SEED_code/DE/session1/'

dir_list = os.listdir(extrac_path)

label = scipy.io.loadmat(label_path)
label = label['label'][0]

for f in dir_list:
    if '_' not in f:
        continue

    S = scipy.io.loadmat(extrac_path + f)
    DE = []
    labelAll = []
    for i in range(15):
        data = S['de_LDS' + str(i + 1)]
        if len(DE):
            DE = np.concatenate((DE, data), axis=1)
        else:
            DE = data

        if len(labelAll):
            labelAll = np.concatenate((labelAll, np.zeros([data.shape[1], 1]) + label[i]), axis = 0)
        else:
            labelAll = np.zeros([data.shape[1], 1]) + label[i]

    #print(DE.shape)
    #print(labelAll.shape)

    mdic = {"DE": DE, "labelAll": labelAll, "label": "experiment"}

    scipy.io.savemat(save_path + f, mdic)
    print(extrac_path + f, '->', save_path + f)


